httpclient/config.go

```diff
@@ -0,0 +1,404 @@
+	cfgKeyRetriesMax                              = "retries.maxAttempts"
+	cfgKeyRetriesPolicyStrategy                   = "retries.policy"
+	cfgKeyRetriesPolicyExponentialInitialInterval = "retries.exponentialBackoff.initialInterval"
+	cfgKeyRetriesPolicyExponentialMultiplier      = "retries.exponentialBackoff.multiplier"
+	cfgKeyRetriesPolicyConstantInternal           = "retries.constantBackoff.interval"
+	cfgKeyRateLimitsEnabled                       = "rateLimits.enabled"
+	cfgKeyRateLimitsLimit                         = "rateLimits.limit"
+	cfgKeyRateLimitsBurst                         = "rateLimits.burst"
+	cfgKeyRateLimitsWaitTimeout                   = "rateLimits.waitTimeout"
+	cfgKeyLogEnabled                              = "log.enabled"
+	cfgKeyLogMode                                 = "log.mode"
+	cfgKeyLogSlowRequestThreshold                 = "log.slowRequestThreshold"
+	cfgKeyMetricsEnabled                          = "metrics.enabled"
+	cfgKeyTimeout                                 = "timeout"
+)
+
+var _ config.Config = (*Config)(nil)
+var _ config.KeyPrefixProvider = (*Config)(nil)
+
+// RateLimitConfig represents configuration options for HTTP client rate limits.
+type RateLimitConfig struct {
+	// Enabled is a flag that enables rate limiting.
+	Enabled bool
+
+	// Limit is the maximum number of requests that can be made.
+	Limit int
+
+	// Burst allow temporary spikes in request rate.
+	Burst int
+
+	// WaitTimeout is the maximum time to wait for a request to be made.
+	WaitTimeout time.Duration
+}
+
+// Set is part of config interface implementation.
+func (c *RateLimitConfig) Set(dp config.DataProvider) (err error) {
+	enabled, err := dp.GetBool(cfgKeyRateLimitsEnabled)
+	if err != nil {
+		return err
+	}
+	c.Enabled = enabled
+
+	if !c.Enabled {
+		return nil
+	}
+
+	limit, err := dp.GetInt(cfgKeyRateLimitsLimit)
+	if err != nil {
+		return err
+	}
+	if limit <= 0 {
+		return dp.WrapKeyErr(cfgKeyRateLimitsLimit, errors.New("must be positive"))
+	}
+	c.Limit = limit
+
+	burst, err := dp.GetInt(cfgKeyRateLimitsBurst)
+	if err != nil {
+		return err
+	}
+	if burst < 0 {
+		return dp.WrapKeyErr(cfgKeyRateLimitsBurst, errors.New("must be positive"))
+	}
+	c.Burst = burst
+
+	waitTimeout, err := dp.GetDuration(cfgKeyRateLimitsWaitTimeout)
+	if err != nil {
+		return err
+	}
+	if waitTimeout < 0 {
+		return dp.WrapKeyErr(cfgKeyRateLimitsWaitTimeout, errors.New("must be positive"))
+	}
+	c.WaitTimeout = waitTimeout
+
+	return nil
+}
+
+// SetProviderDefaults is part of config interface implementation.
+func (c *RateLimitConfig) SetProviderDefaults(_ config.DataProvider) {}
+
+// TransportOpts returns transport options.
+func (c *RateLimitConfig) TransportOpts() RateLimitingRoundTripperOpts {
+	return RateLimitingRoundTripperOpts{
+		Burst:       c.Burst,
+		WaitTimeout: c.WaitTimeout,
+	}
+}
+
+// PolicyConfig represents configuration options for policy retry.
+type PolicyConfig struct {
```

Comment:
Ideally, configuration structs should accurately reflect the configuration structure. For example, if we currently have retries.policy and retries.exponentialBackoff.initialInterval, the corresponding Go structs should look like this:
```go
type RetriesConfig struct {
    Policy             RetryPolicy
    ExponentialBackoff ExponentialBackoffConfig
}

type ExponentialBackoffConfig struct {
    InitialInterval time.Duration
}
```
This approach ensures that the configuration remains clear and aligned with the actual structure.